### PR TITLE
[frontend] Enabled typeahead when optimizer is set to api in the web components

### PR DESCRIPTION
**Pull request by JohanAhlen**

### CODE REVIEW

File: desktop/core/src/desktop/js/apps/editor/components/sqlScratchpad/SqlScratchpad.vue [hunk 1]
```diff
@@ -264,6 +266,7 @@
         logs,
         logsVisible,
         onActiveStatementChanged,
+        sqlAnalyzerRepository,
```
Comment (romainr) [inline desktop/core/src/desktop/js/apps/editor/components/sqlScratchpad/SqlScratchpad.vue:269]: Nice!

File: desktop/core/src/desktop/js/apps/editor/components/aceEditor/acePredict.ts [hunk 2]
```diff
@@ -64,7 +64,9 @@ export const attachPredictTypeahead = (
 
   const updatePredictTypeahead = () => {
     if (activePredictPromise) {
-      activePredictPromise.cancel();
+      try {
+        activePredictPromise.cancel();
+      } catch {}
```
Comment (romainr) [inline desktop/core/src/desktop/js/apps/editor/components/aceEditor/acePredict.ts:69]: Catches to silence potential hic-cups?

**Review by romainr — APPROVED** [review]
Super nice!

File: desktop/core/src/desktop/js/apps/editor/components/aceEditor/acePredict.ts (hunk 2 above)
Comment (JohanAhlen) [reply desktop/core/src/desktop/js/apps/editor/components/aceEditor/acePredict.ts:69]: Indeed, I realized we added the ENABLE_PREDICT flag to silence some errors when not configured properly. We can drop this once it becomes more stable.

**Pull request merged**